src/approaches/oracle_approach.py

```diff
@@ -1001,3 +1010,183 @@ def _get_repeated_nextto_gt_nsrts() -> Set[NSRT]:
     nsrts.add(grasp_nsrt)
 
     return nsrts
+
+
+def _get_behavior_gt_nsrts() -> Set[NSRT]: # pragma: no cover
+    """Create ground truth nsrts for BehaviorEnv."""
+    env = get_env_instance("behavior")
+
+    type_name_to_type = {t.name: t for t in env.types}
+    pred_name_to_pred = {p.name: p for p in env.predicates}
+
+    def _get_lifted_atom(
+        base_pred_name: str, objects: Sequence[Variable]
+    ) -> LiftedAtom:
+        type_names = "-".join(o.type.name for o in objects)
+        pred_name = f"{base_pred_name}-{type_names}"
+        pred = pred_name_to_pred[pred_name]
+        return LiftedAtom(pred, objects)
+
+    def _get_predicate(
+        base_pred_name: str, types: Sequence[Type]
+    ) -> Predicate:
+        type_names = "-".join(t.name for t in types)
+        pred_name = f"{base_pred_name}-{type_names}"
+        return pred_name_to_pred[pred_name]
+
+    # We start by creating nextTo predicates for all possible type
+    # combinations
+    nextto_predicates = set()
+    for next_to_pred_types in itertools.product(env.types, env.types):
+        nextto_predicates.add(_get_predicate("reachable", next_to_pred_types))
+
+    agent_type = type_name_to_type["agent.n.01"]
+    agent_obj = Variable("?agent", agent_type)
+
+    nsrts = set()
+    op_name_count = itertools.count()
+
+    for option in env.options:
+        split_name = option.name.split("-")
+        base_option_name = split_name[0]
+        option_arg_type_names = split_name[1:]
+
+        if base_option_name == "NavigateTo":
+            assert len(option_arg_type_names) == 1
+            target_obj_type_name = option_arg_type_names[0]
+            target_obj_type = type_name_to_type[target_obj_type_name]
+            target_obj = Variable("?targ", target_obj_type)
+
+            # Navigate to from nextto nothing
+            nextto_nothing = _get_lifted_atom("reachable-nothing", [agent_obj])
+            parameters = [agent_obj, target_obj]
+            option_vars = [target_obj]
+            preconditions = {nextto_nothing}
+            add_effects = {
+                _get_lifted_atom("reachable", [target_obj, agent_obj])
+            }
+            delete_effects = {nextto_nothing}
+            nsrt = NSRT(
+                f"{option.name}-{next(op_name_count)}",
+                parameters,
+                preconditions,
+                add_effects,
+                delete_effects,
+                nextto_predicates,
+                option,
+                option_vars,
+                lambda s, r, o: navigate_to_param_sampler(r,\
+                    [env.object_to_ig_object(o_i) for o_i in o], # type: ignore
```

Comment:
why is the type: ignore needed here?

Reply:
ahh I see. okay that's fine, let's just add your answer here as a comment in the code